mlflow/models/model.py

```diff
@@ -799,22 +842,44 @@ def log(
             if serving_input and kwargs.get("validate_serving_input", True):
                 from mlflow.models import validate_serving_input
+                from mlflow.utils.model_utils import env_var_tracker
 
-                try:
-                    validate_serving_input(model_info.model_uri, serving_input)
-                except Exception as e:
-                    _logger.warning(
-                        f"Failed to validate serving input example {serving_input}. "
-                        "Alternatively, you can avoid passing input example and pass model "
-                        "signature instead when logging the model. To ensure the input example "
-                        "is valid prior to serving, please try calling "
-                        "`mlflow.models.validate_serving_input` on the model uri and serving "
-                        "input example. A serving input example can be generated from model "
-                        "input example using "
-                        "`mlflow.models.convert_input_example_to_serving_input` function.\n"
-                        f"Got error: {e}",
-                        exc_info=_logger.isEnabledFor(logging.DEBUG),
-                    )
+                with env_var_tracker() as env:
+                    try:
+                        validate_serving_input(model_info.model_uri, serving_input)
+                    except Exception as e:
+                        _logger.warning(
+                            f"Failed to validate serving input example {serving_input}. "
+                            "Alternatively, you can avoid passing input example and pass model "
+                            "signature instead when logging the model. To ensure the input example "
+                            "is valid prior to serving, please try calling "
+                            "`mlflow.models.validate_serving_input` on the model uri and serving "
+                            "input example. A serving input example can be generated from model "
+                            "input example using "
+                            "`mlflow.models.convert_input_example_to_serving_input` function.\n"
+                            f"Got error: {e}",
+                            exc_info=_logger.isEnabledFor(logging.DEBUG),
+                        )
+                    if hasattr(env, "get_tracked_env_names"):
+                        model_info.env_vars = list(env.get_tracked_env_names()) or None
+            if model_info.env_vars:
+                env_var_path = Path(local_path, ENV_VAR_FILE_NAME)
+                env_var_path.write_text("\n".join(model_info.env_vars) + "\n")
+                mlflow.tracking.fluent.log_artifact(
+                    env_var_path, mlflow_model.artifact_path, run_id
+                )
+                if len(model_info.env_vars) <= 3:
+                    env_var_info = ", ".join(model_info.env_vars)
```

Comment:
```suggestion
                    env_var_info = "[" + ", ".join(model_info.env_vars) + "]"
```

super-nit: to make it a bit more visually standing